pkg/inframetadata/gohai/gohai.go

```diff
@@ -59,13 +64,13 @@ func (m gohaiMarshaler) MarshalJSON() ([]byte, error) {
 // UnmarshalJSON implements the json.Unmarshaler interface.
 // Unmarshals the passed bytes twice (first to a string, then to gohai.Gohai)
 func (m *gohaiMarshaler) UnmarshalJSON(bytes []byte) error {
-	firstUnmarshall := ""
-	err := json.Unmarshal(bytes, &firstUnmarshall)
+	var out string
+	err := json.Unmarshal(bytes, &out)
 	if err != nil {
 		return err
 	}
 
-	err = json.Unmarshal([]byte(firstUnmarshall), &(m.Gohai))
+	err = json.Unmarshal([]byte(out), &(m.Gohai))
```

Comment:
For my own understanding: why cannot it unmarshal `bytes` directly?

Reply:
This comment adds some context https://github.com/DataDog/opentelemetry-mapping-go/blob/1e4c8cc3796bc118ddc2ae31bee54ac60f6bd910/pkg/inframetadata/gohai/gohai.go#L32-L34 for some reason the `gohai` field does not contain a JSON struct, but rather a serialization of a JSON struct into a string.

Reply:
Maybe add some comments here as people who read these lines may have the same doubt

Reply:
Added a couple comments